packages/leancode_hooks/lib/src/use_bloc_listener.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:bloc/bloc.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+/// Takes a [Bloc] or [Cubit] and invokes [listener] in response to state
+/// changes.
+void useBlocListener<S>({
+  required BlocBase<S> bloc,
+  required void Function(S state) listener,
+}) {
+  useEffect(
+    () {
+      final subscription = bloc.stream.listen(listener);
+      return subscription.cancel;
+    },
+    [bloc, listener],
+  );
+}
```

Comment:
Flutter hooks v0.20.0 has a built in stream listener: `useOnStreamChange`

Reply:
But we use 0.18.3